auth/api/iam/api.go

```diff
@@ -628,37 +628,37 @@ func (r Wrapper) OpenIDConfiguration(ctx context.Context, request OpenIDConfigur
 		if errors.Is(err, didsubject.ErrSubjectNotFound) {
 			return nil, oauth.OAuth2Error{
 				Code:        oauth.InvalidRequest,
-				Description: "subject not found",
+				Description: err.Error(),
 			}
 		}
 		return nil, oauth.OAuth2Error{
 			Code:          oauth.ServerError,
 			InternalError: err,
 		}
 	}
-	// resolve DID keys
+	// resolve DID key
+	signingDID := dids[0]
 	set := jwk.NewSet()
 	var signingKey string
-	for _, currentDID := range dids {
-		kid, key, err := r.keyResolver.ResolveKey(currentDID, nil, resolver.AssertionMethod)
-		if err != nil {
-			return nil, oauth.OAuth2Error{
-				Code:          oauth.ServerError,
-				InternalError: err,
-			}
+	kid, key, err := r.keyResolver.ResolveKey(signingDID, nil, resolver.AssertionMethod)
+	if err != nil {
+		return nil, oauth.OAuth2Error{
+			Code:          oauth.ServerError,
+			InternalError: err,
 		}
-		// create JWK and add to set
-		jwkKey, err := jwk.FromRaw(key)
-		if err != nil {
-			return nil, oauth.OAuth2Error{
-				Code:          oauth.ServerError,
-				InternalError: err,
-			}
+	}
+	// create JWK and add to set
+	jwkKey, err := jwk.FromRaw(key)
+	if err != nil {
+		return nil, oauth.OAuth2Error{
+			Code:          oauth.ServerError,
+			InternalError: err,
 		}
-		_ = jwkKey.Set(jwk.KeyIDKey, kid)
-		_ = set.AddKey(jwkKey)
-		signingKey = kid
 	}
+	_ = jwkKey.Set(jwk.KeyIDKey, kid)
+	_ = set.AddKey(jwkKey)
```

Comment:
`set` needs to contain all keys in `dids`

Reply:
aah, now I udnerstand Wout's comment